WordPress/src/test/java/org/wordpress/android/ui/jetpack/scan/details/ThreatDetailsViewModelTest.kt

```diff
@@ -11,51 +11,213 @@ import org.junit.Test
     @Mock private lateinit var builder: ThreatDetailsListItemsBuilder
+    @Mock private lateinit var htmlMessageUtils: HtmlMessageUtils
+    @Mock private lateinit var resourceProvider: ResourceProvider
     private lateinit var viewModel: ThreatDetailsViewModel
     private val threatId = 1L
+    private val fakeUiStringText = UiStringText("")
 
     @Before
-    fun setUp() {
-        viewModel = ThreatDetailsViewModel(getThreatModelUseCase, builder)
+    fun setUp() = test {
+        viewModel = ThreatDetailsViewModel(
+            getThreatModelUseCase,
+            ignoreThreatUseCase,
+            selectedSiteRepository,
+            builder,
+            htmlMessageUtils,
+            resourceProvider,
+            TEST_DISPATCHER
+        )
+        whenever(site.name).thenReturn(TEST_SITE_NAME)
+        whenever(selectedSiteRepository.getSelectedSite()).thenReturn(site)
+        whenever(htmlMessageUtils.getHtmlMessageFromStringFormatResId(any(), any())).thenReturn(mock())
+        whenever(getThreatModelUseCase.get(anyLong())).thenReturn(mock())
+        whenever(builder.buildThreatDetailsListItems(any(), any(), any(), any())).thenAnswer {
+            createDummyThreatDetailsListItems(it.getArgument(ON_IGNORE_THREAT_BUTTON_CLICKED_PARAM_POSITION))
+        }
     }
 
     @Test
     fun `given threat id, when on start, then threat details are retrieved`() = test {
-        // Act
         viewModel.start(threatId)
-        // Assert
+
         verify(getThreatModelUseCase).get(threatId)
     }
 
     @Test
     fun `given threat id, when on start, then ui is updated with content`() = test {
-        // Arrange
         val uiStates = init().uiStates
-        // Act
-        viewModel.start(threatId)
-        // Assert
+
         val uiState = uiStates.last()
         assertThat(uiState).isInstanceOf(Content::class.java)
     }
 
-    private suspend fun init(): Observers {
+    @Test
+    fun `when ignore threat button is clicked, then action to open ignore action confirmation dialog is triggered`() =
+        test {
+            val expectedDialogTitle = UiStringRes(R.string.threat_ignore)
+            val expectedDialogMessage = UiStringText(
+                htmlMessageUtils
+                    .getHtmlMessageFromStringFormatResId(
+                        R.string.threat_ignore_warning,
+                        "<b>${site.name ?: resourceProvider.getString(R.string.scan_this_site)}</b>"
+                    )
+            )
+            val expectedPositiveButtonLabel = R.string.dialog_button_ok
+            val expectedNegativeButtonLabel = R.string.dialog_button_cancel
+            val observers = init()
+
+            (observers.uiStates.last() as Content).items.filterIsInstance<ActionButtonState>()
+                .first().onClick.invoke()
+
+            val confirmationDialogAction = observers.navigation.last().peekContent() as OpenThreatActionDialog
```

Comment:
💡 Idea: We might want to consider splitting this method into 2 methods. 

First method would assert that ` observers.navigation.last().peekContent()` is instance of `OpenThreatActionDialog` == `when ignore threat button is clicked, then action to open ignore action confirmation dialog is triggered`.

Second method would assert that the dialog shows the expected content.

I agree that splitting it into 5 methods (assert title, message, positive btn, .. ) is an overkill, but it feels like the current method tests two different things - correct navigation + correct content. Wdyt?